### PR TITLE
fix: [CDS-77433]: supporting moduleConfig expression for terragrunt plan, apply and destroy

### PR DESCRIPTION
**Pull request by namantalaycha10**
## Describe your changes
supporting this expression <+execution.steps.TerragruntPlan_1.spec.configuration.moduleConfig.path> which is not resolving right now 
there is already another expression for the same <+execution.steps.TerragruntPlan_1.spec.configuration.terragruntModuleConfig.path>   which is right now used by the customer in place of the first one

Before changes: 
image

After changes:
image

Screenshot 2023-08-25 at 7 15 54 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52015)

### CODE REVIEW

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by yogesh-chauhan** [comment]
@namantalaycha10 could you please update the PR description ? Thanks!

**Comment by vlica-harness** [comment]
Hi @yogesh-chauhan , do we include the fix in this PR not showing both terragruntModuleConfig and moduleConfig in inputs as you suggested in this area? io.harness.cdng.service.steps.helpers.beans.ServiceStepV3Parameters#toViewJson

**Comment by yogesh-chauhan** [comment]
Yes, we should fix it in this PR itself @namantalaycha10

**Comment by namantalaycha10** [comment]
trigger uts

**Comment by namantalaycha10** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terragrunt/TerragruntExecutionDataParameters.java [hunk 1]
```diff
@@ -27,6 +30,7 @@
 import lombok.Builder;
 import lombok.Data;
 
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_K8S})
```
Comment (vlica-harness) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terragrunt/TerragruntExecutionDataParameters.java:33]: it this CDS_K8S HarnessModuleComponent correct? or we need to have smth la infrastructure or terragrunt spefic?

Comment (namantalaycha10) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/provision/terragrunt/TerragruntExecutionDataParameters.java]: done

**Comment by namantalaycha10** [comment]
trigger uts

**Review by vlica-harness — APPROVED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 6 files at r1, 7 of 7 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52015)* status: 9 file(s) to review. 1. unresolved discussions.

**Comment by yogesh-chauhan** [comment]
trigger review

**Comment by namantalaycha10** [comment]
trigger messagecheck

**Pull request merged**